docs/authentication.md

```diff
@@ -0,0 +1,151 @@
+ * `soft_expire`  
+    (in seconds) When a user is authenticated, a session is created for them and this session identifies the user's access.
+    This configuration variable sets how long the session considered "valid" before the user is needed
+    to reauthenticate again &mdash; if this time expires, the session will be *hibernated*: the next access will be denied,
+    but if the user presents a valid login, they will get their session reused.
+ * `session_lifetime`  
+    (in seconds) The lifetime of the session sets that after this many seconds since last session access the session is permanently invalidated.
+
+Every authentication method is its own JSON object in this section. Every authentication method has its
+own `enabled` key which dictates whether it is used at live authentication or not.
+
+Users are authenticated if **any** authentication method successfully authenticates them. If both methods are enabled, *dictionary* authentication takes precedence.
+
+### *Dictionary* authentication
+
+The `authentication.method_dictionary` contains a plaintext `username:password` credentials for authentication.
+If the user's login matches any of the credentials listed, the user will be authenticated.
+
+```json
+"method_dictionary": {
+  "enabled" : true,
+  "auths" : [
+      "global:admin",
+      "test:test"
+  ]
+}
+```
+
+### *LDAP* authentication
+
+CodeChecker also supports *LDAP*-based authentication. The `authentication.method_ldap` section contains the configuration for LDAP authentication:
+the server can be configured to connect to as much LDAP-servers as the administrator wants. Each LDAP server is identified by a `connection_url` and a list of `queries`
+to attempt to log in the username given.
+
+Servers are connected to and queries are executed in the order they appear in the configuration file.
+Because of this, it is not advised to list too many servers as it can elongenate the authentication process.
+
+The special `$USN$` token in the query is replaced to the *username* at login.
+
+```json
+"method_ldap": {
+  "enabled" : true,
+  "authorities": [
+    {
+      "connection_url": "ldap://ldap.example.org",
+      "queries": [
+        "uid=$USN$,ou=admins,o=mycompany"
+      ]
+    },
+    {
+      "connection_url" : "ldaps://secure.internal.example.org:636",
+      "queries": [
+        "uid=$USN$,ou=owners,ou=secure,o=company"
+      ]
+    }
+  ]
+}
+```
+
+----
+
+## Clientside configuration
+
+### Web-browser client
+
+Authentication in the web browser is handled via standard *HTTP Authenticate* headers, the browser will prompt the user to supply their crendentials.
+
+For browser authentication to work, cookies must be enabled!
+
+### Command-line client
+
+The `CodeChecker cmd` client needs to be authenticated for a server before any data communication could take place.
+
+The client's configuration file is expected to be at `~/.codechecker_passwords.json`, which is created at the first command executed
```

Comment:
:smirk: Better idea is to create the file initially as `0600`. Though I added it to the documentation and a warning (in `ssh` style) for the user if the file is with the wrong permissions.